### DISCUSSION TITLE
Add Perfect FLAC definition as used by RED

### DISCUSSION THREAD

**Discussion by frantathefranta** [autobrr/autobrr#1688] [Ideas]
**Is your feature request related to a problem? Please describe.**
When using a Music oriented filter, the helpful "Perfect FLAC" override is different in definition than what the Redacted tracker uses. Redacted states that "(100% log for CD, or any Vinyl/DVD/Soundboard/WEB/Cassette/Blu-ray/SACD/DAT)" counts for a Perfect FLAC. I realize that autobrr uses the Orpheus definition, which I respect. 

**Describe the solution you'd like**
I'd like to have an option to select the Redacted definition.

**Describe alternatives you've considered**
If it's not possible to do this, I'd like to know how to set up a filter that covers the Redacted definition of Perfect FLAC, as when I try to select it manually in the simple toggles, I can't get it to snatch both WEB FLACs and CD FLACs with 100% log.

**Comment by zze0s** [comment]
Hi! I think we use the varroa definition of Perfect FLAC which is CD, FLAC, Lossless/24bit lossless, Cue file, Log file, Log Score 100%.

You should be able to setup a filter for that:

```JSON
{
    "name": "Music",
    "version": "1.0",
    "data": {
        "formats": [
            "FLAC"
        ],
        "quality": [
            "Lossless",
            "24bit Lossless"
        ],
        "media": [
            "CD",
            "WEB",
            "Vinyl",
            "Soundboard",
            "DAT",
            "Cassette",
            "Blu-Ray",
            "SACD"
        ],
        "cue": true,
        "log": true,
        "log_score": 100
    }
}
```

We can look into if the predefined Perfect FLAC setting should be changed, but I am a bit hesitant as there are a lot of users relying on it that might start to get unwanted releases after a change like that.

**Comment by frantathefranta** [comment]
Yeah I definitely don't want to break it for anyone. I feel like I've tried this and it didn't grab Lossless WEB files, but I'll try again.

**Comment by zze0s** [comment]
Please do and set your log level in settings to either Debug or Trace for like a day and see what it does.

**Comment by marlowe-p** [comment]
I encountered this same situation. My understanding from the docs and the filter definition is that it's behaving as expected, but I would also find it helpful to somehow enable the "perfect FLAC" definition from RED. As it is now, my workaround is to have two filters: one for perfect FLAC (CD with cue and 100% log) and one for FLAC WEB/Vinyl/Cassette. But this is not sustainable because I need to duplicate the filtering conditions, and for example adding one tag means adding it to two filters, which is difficult to maintain.

I have some recent logs that confirm the current behaviour is as OP describes: the "perfect FLAC"  toggle doesn't fetch, say, WEB releases.
```
01:06:24 DBG checking filter: Music (electronic – exploration) with release phase trois - Magic Maze II [2026] (WEB FLAC Lossless) method=CheckFilter module=filter
01:06:24 DBG (Music (electronic – exploration)) for release: phase trois - Magic Maze II [2026] (WEB FLAC Lossless) rejections: ([perfect flac] not matching: got [FLAC Lossless] want: Cue, Log, Log Score 100, FLAC and 24bit Lossless) method=CheckFilter module=filter
01:06:24 DBG filter Music (electronic – exploration) rejected release: phase trois - Magic Maze II [2026] (WEB FLAC Lossless) with reasons: [perfect flac] not matching: got [FLAC Lossless] want: Cue, Log, Log Score 100, FLAC and 24bit Lossless filter="Music (electronic – exploration)" indexer=redacted module=release release="phase trois - Magic Maze II [2026] (WEB FLAC Lossless)"
01:06:24 DBG checking filter: Music (electronic – exploration – WEB) with release phase trois - Magic Maze II [2026] (WEB FLAC Lossless) method=CheckFilter module=filter
01:06:25 INF Matched 'phase trois - Magic Maze II [2026] (WEB FLAC Lossless)' (Music (electronic – exploration – WEB)) for Redacted filter="Music (electronic – exploration – WEB)" indexer=redacted module=release release="phase trois - Magic Maze II [2026] (WEB FLAC Lossless)"
```

**Reply by zze0s** [reply]
What is the exact definition of Perfect FLAC at RED, and OPS?

**Reply by zze0s** [reply]
Take a look at https://github.com/autobrr/autobrr/pull/2534 . It should have current definition of what both RED and OPS define as Perfect FLAC.

**Reply by marlowe-p** [reply]
That's awesome! Thank you kindly for addressing this so quickly.

**Reply by marlowe-p** [reply]
@zze0s I took the liberty of creating [this small PR](https://github.com/autobrr/autobrr.com/pull/380) to update the docs to reflect your changes.

**Comment by marlowe-p** [comment]
It seems to me the problem stems from the fact that the conditions
```
"cue": true,
"log": true,
"log_score": 100
```
are checked regardless of the media, when in fact they only make sense if the media is CD. For example, a WEB release will never have a log file, so rejecting the release because it lacks a log file seems unreasonable to me.
